rules/ruleset.go

```diff
@@ -42,18 +42,18 @@ var (
 )
 
 // MatchMode determines how match is performed.
-type MatchMode int
+type MatchMode string
 
 // List of supported match modes.
 const (
 	// When performing matches in ForwardMatch mode, the matcher matches the given id against
 	// both the mapping rules and rollup rules to find out the applicable mapping policies
 	// and rollup policies.
-	ForwardMatch MatchMode = iota
+	ForwardMatch MatchMode = "forward"
```

Comment:
Can we instead implement `UnmarshalYAML` for `MatchMode`? This way it's more standardized and we don't need to add custom parsing logic elsewhere.